### ISSUE TITLE
[Enhancement] show loading spinner when applying load order

### ISSUE THREAD

**Issue by SuperSandro2000** [loot/loot#878]
When you have a long load order and want to apply it after sorting it sometimes the applying takes several seconds to almost half a minute for me. Maybe there could be added a little spinner to inform the user that the operation is running.

**Comment by WrinklyNinja** [comment]
It should be nearly instantaneous, please post your debug log.

**Comment by SuperSandro2000** [comment]
```
[12:58:20.416284] [trace]: User has accepted sorted load order, applying it.
[12:58:20.416355] [debug]: Getting load order.
[12:58:20.426212] [info]: Setting load order.
[12:58:20.426236] [info]: 		FalloutNV.esm
.......
[12:58:20.427170] [info]: 		TTW Realistic Wasteland Lighting.esp
[12:58:43.812319] [info]: Load order set successfully.

```
My load order is 120 items long. Hope this is not the cause.

**Comment by WrinklyNinja** [comment]
Your full debug log, please. Just upload it as an attachment.

**Comment by SuperSandro2000** [comment]
sry, thought this would be enough
[LOOTDebugLog.txt](https://github.com/loot/loot/files/1714407/LOOTDebugLog.txt)

**Comment by WrinklyNinja** [comment]
Do you by any chance have a bunch of loose files in your Data folder?

**Comment by SuperSandro2000** [comment]
I am not at my pc right now but I think yes. I use mo2 and some mods like nmc and a few bigger quest mods for fnv and f3(ttw) are lose. But why does this matter? Doesn't loot only touch .esp/.esm/.esl files?

**Comment by WrinklyNinja** [comment]
> Doesn't loot only touch .esp/.esm/.esl files?

LOOT only parses them, but I've just realised it's scanning the data directory for them recursively, which is unnecessary, and probably explains why you see getting and setting the load order taking 20-30 seconds and I see it it taking < 20 ms.

Once it's done, I'll link to a build with that fixed for you to try.

**Comment by WrinklyNinja** [comment]
OK, replace the `loot_api.dll` in your LOOT install with [this one](https://bintray.com/wrinklyninja/loot/download_file?file_path=loot_api-0.12.3-4-g627c64b_update-deps-win32.7z) and try again.

**Comment by SuperSandro2000** [comment]
Applying takes now like one second. I am not sure but the other processes feel a little bit fast too.
[LOOTDebugLog.txt](https://github.com/loot/loot/files/1715627/LOOTDebugLog.txt)

**Comment by WrinklyNinja** [comment]
From your logs it's actually now ~ 70 ms seconds. A ~ 500x improvement isn't bad, but that's still pretty slow. My new benchmarking (WrinklyNinja/libloadorder#44) suggests it can be at least 10x faster, I'll see what I can do.

**Comment by SuperSandro2000** [comment]
Hit me up if you have something to test.

**Comment by WrinklyNinja** [comment]
Well, I don't think I could get it 10x faster without compromising correctness, but try using [this loot_api.dll](https://bintray.com/wrinklyninja/loot/download_file?file_path=loot_api-0.12.3-4-gf28eda8_update-deps-win32.7z).

**Comment by SuperSandro2000** [comment]
I test it tomorrow.
But I noticed that sometimes loot does not probably sort esm and esp files. I look if I can reliable reproduce it but probably a derp on my end.

**Comment by SuperSandro2000** [comment]
The sorting problem is proabably just me forgetting hitting sort... upsi.
Applying is now almost instant but I changed my setup and currently my load order is shorter but still way faster than before. I attached a Debug Log for reference.
[LOOTDebugLog.txt](https://github.com/loot/loot/files/1727233/LOOTDebugLog.txt)

**Comment by WrinklyNinja** [comment]
I count a three plugin difference (2% of your load order) and a 50% speed increase, so that's good.

I'll leave this issue open until I've released a new version of the LOOT API and updated LOOT to depend on it, but I don't think it needs any further work otherwise.

**Comment by Al12rs** [comment]
@WrinklyNinja Do you have an ETA for the LOOT api release containing this fix? We wanted to make a Mod Organizer 2.1.2 release and including this fix seems sensible.

**Comment by WrinklyNinja** [comment]
I can get a release out tomorrow.

**Comment by WrinklyNinja** [comment]
LOOT API v0.12.4 has been [released](https://github.com/loot/loot-api/releases/tag/0.12.4).LOOT API v0.12.4 has been [released](https://github.com/loot/loot-api/releases/tag/0.12.4).

**Comment by WrinklyNinja** [comment]
Merged into `dev` as of 030e91d468d5e410cbc5ed363f8f5b18c4e5cd31, so closing this.
